ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java

```diff
@@ -0,0 +1,281 @@
+      StoreKeyFactory storeKeyFactory, ClusterMap clusterMap, Time time) {
+    this.dataNodeId = dataNodeId;
+    this.fetchSizeInBytes = fetchSizeInBytes;
+    this.storeConfig = storeConfig;
+    this.storeKeyFactory = storeKeyFactory;
+    this.clusterMap = clusterMap;
+    this.time = time;
+    consistencyChecker =
+        new ConsistencyCheckerTool(clusterMap, new StoreToolsMetrics(clusterMap.getMetricRegistry()), time);
+  }
+
+  /**
+   * Performs a reformat of the disk.
+   * 1. Copies one partition on the disk to a scratch space
+   * 2. Performs local copies of all other partitions on the disk using {@link StoreCopier} and deletes the source.
+   * 3. Copies the partition in the scratch space back on to the disk
+   * 4. Deletes the folder in the scratch space
+   * @param diskMountPath the mount path of the disk to reformat
+   * @param scratch the scratch space to use
+   * @throws Exception
+   */
+  public void reformat(String diskMountPath, File scratch) throws Exception {
+    if (!scratch.exists()) {
+      throw new IllegalArgumentException("Scratch space " + scratch + " does not exist");
+    }
+    List<ReplicaId> replicasOnDisk = new ArrayList<>();
+    // populate the replicas on disk
+    List<? extends ReplicaId> replicaIds = clusterMap.getReplicaIds(dataNodeId);
+    for (ReplicaId replicaId : replicaIds) {
+      if (replicaId.getDiskId().getMountPath().equals(diskMountPath)) {
+        replicasOnDisk.add(replicaId);
+      }
+    }
+    if (replicasOnDisk.size() == 0) {
+      throw new IllegalArgumentException("There are no replicas on " + diskMountPath + " of " + dataNodeId);
+    }
+    replicasOnDisk.sort(Comparator.comparingLong(ReplicaId::getCapacityInBytes));
+    logger.info("Found {} on {}", replicasOnDisk, diskMountPath);
+
+    // move the last replica id (the largest one) to scratch space
+    ReplicaId toMove = replicasOnDisk.get(replicasOnDisk.size() - 1);
+    String partIdString = toMove.getPartitionId().toString();
+    File scratchSrc = new File(toMove.getReplicaPath());
+    File scratchTgt = new File(scratch, partIdString + RELOCATED_DIR_NAME_SUFFIX);
+    if (scratchTgt.exists()) {
+      throw new IllegalStateException(scratchTgt + " already exists");
+    }
+    logger.info("Moving {} to {}", scratchSrc, scratchTgt);
+    delete(scratchTgt);
+    FileUtils.moveDirectory(scratchSrc, scratchTgt);
+
+    // reformat each store, except the one moved, one by one
+    for (int i = 0; i < replicasOnDisk.size() - 1; i++) {
+      ReplicaId replicaId = replicasOnDisk.get(i);
+      partIdString = replicaId.getPartitionId().toString();
+      File src = new File(replicaId.getReplicaPath());
+      File tgt = new File(replicaId.getMountPath(), partIdString + UNDER_REFORMAT_DIR_NAME_SUFFIX);
+      logger.info("Copying {} to {}", src, tgt);
+      copy(partIdString, src, tgt, replicaId.getCapacityInBytes());
+      delete(src);
```

Comment:
Discussed this offline - there doesn't seem to be an atomic move directory. The articles I read online say that move dir methods also may delete and move, so will leave it this way for now. Looking to see if there are other options.